src/server/config/cmds.go

```diff
@@ -0,0 +1,305 @@
+			}
+
+			context, err := readContext()
+			if err != nil {
+				return err
+			}
+
+			if !overwrite {
+				if _, ok := cfg.V2.Contexts[args[0]]; ok {
+					return fmt.Errorf("context '%s' already exists, use `--overwrite` if you wish to replace it", args[0])
+				}
+			}
+
+			cfg.V2.Contexts[args[0]] = context
+			return cfg.Write()
+		}),
+	}
+	setContext.Flags().BoolVar(&overwrite, "overwrite", false, "Overwrite a context if it already exists.")
+	commands = append(commands, cmdutil.CreateAlias(setContext, "config set context"))
+
+	var pachdAddress string
+	updateContext := &cobra.Command{
+		Short: "Updates a context.",
+		Long:  "Updates an existing context config from a given name.",
+		Run: cmdutil.RunCmdFixedArgs(1, func(cmd *cobra.Command, args []string) (retErr error) {
+			cfg, err := config.Read()
+			if err != nil {
+				return err
+			}
+
+			context, ok := cfg.V2.Contexts[args[0]]
+			if !ok {
+				return fmt.Errorf("context does not exist: %s", args[0])
+			}
+
+			if cmd.Flags().Changed("pachd-address") {
+				// Use this method since we want to differentiate between no
+				// `pachd-address` flag being set (the value shouldn't be
+				// changed) vs the flag being an empty string (meaning we want
+				// to set the value to an empty string)
+				context.PachdAddress = pachdAddress
+			}
+
+			return cfg.Write()
+		}),
+	}
+	updateContext.Flags().StringVar(&pachdAddress, "pachd-address", "", "Set a new name pachd address.")
+	commands = append(commands, cmdutil.CreateAlias(updateContext, "config update context"))
+
+	deleteContext := &cobra.Command{
+		Short: "Deletes a context.",
+		Long:  "Deletes a context.",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) (retErr error) {
+			cfg, err := config.Read()
+			if err != nil {
+				return err
+			}
+			if _, ok := cfg.V2.Contexts[args[0]]; !ok {
+				return fmt.Errorf("context does not exist: %s", args[0])
+			}
+			if cfg.V2.ActiveContext == args[0] {
+				return errors.New("cannot delete an active context")
+			}
+			delete(cfg.V2.Contexts, args[0])
+			return cfg.Write()
+		}),
+	}
+	commands = append(commands, cmdutil.CreateAlias(deleteContext, "config delete context"))
+
+	listContext := &cobra.Command{
+		Short: "Lists contexts.",
+		Long:  "Lists contexts.",
+		Run: cmdutil.Run(func(args []string) (retErr error) {
+			cfg, err := config.Read()
+			if err != nil {
+				return err
+			}
+
+			keys := make([]string, len(cfg.V2.Contexts))
+			i := 0
+			for key := range cfg.V2.Contexts {
+				keys[i] = key
+				i++
+			}
+			sort.Strings(keys)
+
+			fmt.Println(listContextHeader)
+			for _, key := range keys {
+				if key == cfg.V2.ActiveContext {
+					fmt.Printf("*\t%s\n", key)
+				} else {
+					fmt.Printf("\t%s\n", key)
+				}
+			}
+			return nil
+		}),
+	}
+	commands = append(commands, cmdutil.CreateAlias(listContext, "config list context"))
+
+	// TODO(ys): help template and usage func are nearly a copy-paste from
+	// what's in cmdutil/cobra.go, clean this up
```

Comment:
@Tryneus this is kind of gnarly; it's a copy-paste of what's cmdutil/cobra.go, and the way the walk function finds associated commands is super hackey. If you've got ideas for how to DRY this up or otherwise clean it up, let me know.